api/app/utils/untappd.ts

```diff
@@ -36,44 +42,90 @@ export async function getBeerInfo(beerId: number): Promise<BeerInfo> {
             .parameter('users', TYPES.NVarChar, usersList)
             .executeImmediate();
-        return await Promise.all(sessions
-            .map((session: Session) => { return {Session: session, User: users.find((user) => session.PersonnelNumber === user.PersonnelNumber)}; })
+        var retVal = await sessions
+            .map((session: Activity) => { return { Session: session, User: users.find((user) => session.PersonnelNumber === user.PersonnelNumber) }; })
             .filter(session => session.User && session.Session.UntappdId)
-            .map(session => { 
+            .mapAsync(async session => {
                 var params = {
-                    client_id: process.env.UntappdClientId,
-                    client_secret: process.env.UntappdClientSecret,
-                    access_token: session.User.UntappdAccessToken,
                     gmt_offset: new Date().getTimezoneOffset() / 60,
+                    timezone: process.env.Timezone,
                     bid: session.Session.UntappdId,
+                    foursquare_id: process.env.FourSquareVenueId,
+                    geolat: process.env.VenueLat,
+                    geolng: process.env.VenueLong,
                     facebook: session.User.CheckinFacebook,
                     twitter: session.User.CheckinTwitter,
                     foursquare: session.User.CheckinFoursquare
                 }
-                var checkIn = request_promise.post({
-                    uri: `${untappdApiRoot}checkin/add?` + Object.keys(params).map(param => `${param}=${params[param]}`).join('&'),
-                    json: true
+
+                try {
+                    var checkin = await request_promise.post({
+                        uri: `${untappdApiRoot}checkin/add?access_token=` + session.User.UntappdAccessToken,
+                        form: params
+                    });
+
+                    var checkinResp = JSON.parse(checkin);
+                    return { activityId: session.Session.SessionId, untappdCheckin: checkinResp.response }
+                }
+                catch (ex) {
+                    return null;
+                }
+            });
+
+        if (!!retVal && retVal.length > 0) {
+            sqlStatement = "UPDATE dbo.[FactDrinkers] " +
+                "SET [UntappdCheckinId]=@checkinId, " +
+                "[UntappdBadgeName]=@badgeName, " +
+                "[UntappdBadgeImageURL]=@imageUrl " +
+                "WHERE [Id]=@activityId";
+
+            var connection = new tds.TdsConnection();
+            var updateUntappdCheckin = await connection.sql(sqlStatement)
+                .parameter('checkinId', TYPES.Int, null)
+                .parameter('badgeName', TYPES.NVarChar, null, { length: 500 })
+                .parameter('imageUrl', TYPES.NVarChar, null, { length: 1000 })
+                .parameter('activityId', TYPES.Int, null)
+                .prepare();
+
+            await retVal.forEachAsync(async activity => {
```

Comment:
Do you need to handle the case where the checkin didn't work (returned null from the local catch handler) and therefore the activity lambda argument is null?